### PR TITLE
Concurrent downstream

### PR DESCRIPTION
**Pull request by amimimor**
# Description

based on metadata property: `concurrencyMode` call subscriber' handler either concurrently or one message at a time (sequentially, as it is before this PR)

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1650 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [X] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#2354

### CODE REVIEW

**Review by berndverst — APPROVED** [review]

**Comment by berndverst** [comment]
Thanks for the contribution. Please open a docs issue and a PR to make the necessary docs change to document this metadata field, then link the PR here.

Once you have opened a PR against docs we can merge this.

**Comment by amimimor** [comment]
> Thanks for the contribution. Please open a docs issue and a PR to make the necessary docs change to document this metadata field, then link the PR here.
> 
> 
> 
> Once you have opened a PR against docs we can merge this.

Sure. The PR of the docs should be for v1.7?

**Comment by berndverst** [comment]
@amimimor please make the PR against docs `v1.8` branch because this change will not appear in Dapr until the v1.8.0 release. Thanks.

**Comment by amimimor** [comment]
> @amimimor please make the PR against docs `v1.8` branch because this change will not appear in Dapr until the v1.8.0 release. Thanks.

done. see: https://github.com/dapr/docs/issues/2354 and its associated PR

**Comment by amimimor** [comment]
hi @berndverst, we have internally discussed this feature and I want to further improve the implementation so I please don't merge this PR yet. I can close this PR if you think it is more appropriate, rather than wait for my further changes to be introduced. Waiting for your call on this.

File: pubsub/aws/snssqs/metadata.go
Comment (amimimor) [reply pubsub/aws/snssqs/metadata.go:158]: Yes. That's true for every component using that pubsub metadata field. I mean that deciding that "parallel" is the default is the project decision. You think we should change that? Document that?

Comment (msfussell) [reply pubsub/aws/snssqs/metadata.go:158]: Does this concurrency metadata field affect multiple (all?) pub/sub components?

Comment (berndverst) [reply pubsub/aws/snssqs/metadata.go:158]: No this only affects this component. It however uses a helper function that scans metadata to look for a concurrency value and returns "parallel" if it can't find one.

The problem is that before this change the SNS/SQS component will likely have used a different default concurrency value.

Maybe we should just duplicate the helper function here within the component, but without specifying a default value so we can retain the existing behavior (whatever that even is).

Comment (yaron2) [reply pubsub/aws/snssqs/metadata.go:158]: The existing behavior is very clearly single concurrency mode.

Comment (amimimor) [reply pubsub/aws/snssqs/metadata.go:158]: AFAIR the Rabbit MQ component also uses this construct. I'm totally fine with setting the "single" mode as default although I'm not too sure if that's what most users would expect (we were surprised at first) because most webservers and languages are concurrent out of the box nowadays. I also want to remind that one of the advantages of SQS is batching of message consumption and therefore sending the batch concurrently to the handler fits the idea nicely

Comment (yaron2) [reply pubsub/aws/snssqs/metadata.go:158]: I think we should enable `parallel` as the default for SQS and explicitly mention it on the documentation (and also as a highlighted change in the 1.8 release notes). As a Beta component we can make this change now, also this doesn't change the APIs so I feel comfortable with this.

**Review by berndverst — APPROVED** [review]

**Comment by berndverst** [comment]
Docs PRs and Release notes issue look good. Merging this.

**Pull request merged**
